added a info about the auto heal of k8 (#49282)
kubernetes/website: content/en/docs/concepts/architecture/_index.md

{{< glossary_definition term_id="container-runtime" length="all" >}}

## Ability to self-heal

The idea behind self-healing Kubernetes is simple: If a container fails, Kubernetes automatically redeploys the afflicted 

> container to its desired state to restore operations

Reviewer: ```suggestion
container to its desired state to restore operations.
```

> Self-healing Kubernetes has four capabilities :

Reviewer: ```suggestion
Self-healing Kubernetes has four capabilities:
```

 - restart failed containers.
 - replace containers that require updates, such as a new software version.
 - disable containers that don't respond to predefined health checks.
 - prevent containers from appearing to users or other containers until they are ready.

> 

Reviewer 2: - Restart failed containers.
- Replace containers that require updates, such as a new software version.
- Kill containers that don't respond to user-defined health checks.
- Prevent advertising containers to clients until they are ready to serve traffic.

> Ideally, container detection and restoration should be seamless and immediate, minimize application disruption and mitigate negative UX.

Reviewer: ```suggestion
Ideally, container detection and restoration should be seamless and immediate, minimize application disruption and mitigate negative user experience.
```

Organizations can specify how Kubernetes performs health checks and what actions it should take after it detects a problem.

> How does self-healing work with Kubernetes? :

Reviewer: ```suggestion
How does self-healing work with Kubernetes?:
```
